### PR TITLE
Enable leaf project restore from root build

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/corefx/issues/40052

cc @Wraith2

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Removed dirs.proj as well as that code belongs into build.proj and we can remove another consumer of dir.traversal.targets.

File: Directory.Build.props [hunk 1]
```diff
@@ -333,9 +333,9 @@
     <PackageConflictPlatformManifests Include="$(PlatformManifestFile)" />
   </ItemGroup>
 
-  <!-- AOT only asset that we currently don't support. -->
```
Comment (ericstj) [inline Directory.Build.props]: nit: appears unrelated to this change.

File: build.proj [hunk 2]
```diff
@@ -3,9 +3,9 @@
   <Import Project="Directory.Build.props" />
 
   <PropertyGroup>
-    <BuildNative Condition="'$(BuildNative)' == ''">true</BuildNative>
+    <BuildNative Condition="'$(BuildNative)' == '' and '$(DirectoryToBuild)' == ''">true</BuildNative>
```
Comment (ericstj) [inline build.proj]: Observation: DirectoryToBuild appears  14 times in this commit.  Wish we didn't need to special case something that's only excercised through manual interaction so much.

Comment (ericstj) [reply build.proj]: Also, where is DirectoryToBuild documented along with how it's supposed to be used?  What real value does it provide over someone just directly building a project?

File: build.proj [hunk 3]
```diff
@@ -39,11 +39,28 @@
   </Target>
 
   <Target Name="BuildManaged">
-    <ItemGroup>
-      <_BuildMangedProjects Include="src\dirs.proj" />
+    <ItemGroup Condition="'$(DirectoryToBuild)' == ''">
```
Comment (ericstj) [inline build.proj]: Was there a specific reason you moved this into build.proj?

Comment (ViktorHofer) [reply build.proj]: Dirs.proj was a container before but as tests are handled separately it only handled the managed non-test part of the build. As we don't have such partial containers I simply moved it to build.proj where it better fits IMO.

File: build.proj (hunk 2 above)
Comment (ViktorHofer) [reply build.proj]: We need to document DirectoryToBuild properly I think. The biggest advantage of it is that it allows building a leaf project without the need to invoke msbuild manually. This is very convenient on machines where you don't have a fitting global SDK installed and don't want to set env vars to point to the local one.

**Comment by ViktorHofer** [comment]
Removing the build.proj file unfortunately leads to issues with the configuration system and the FilterProjects target. I undid that change.

**Comment by ViktorHofer** [comment]
Merging as the change is fine and Eric already took a look. I addressed one feedback item and the other one remains but I'm not planning to change the DirectoryToBuild feature at this point. With changes in the configuration system it might be easier to point the arcade Build.proj directly to the specified project/sln but I don't think that's currently possible.

File: build.proj (hunk 2 above)
Comment (ericstj) [reply build.proj]: It just seems rather odd to me that the only thing you can specify for DirectoryToBuild is something like src\foo, there are odd assumptions about the substructure of the library folders that aren't really all that useful.  Added to that the footprint of this feature across a bunch of different files and it led me to ask this question about how its supposed to work and what's the real value.
